docs/plugins.md

```diff
@@ -19,22 +19,29 @@ There are two main components that specify plugin behavior:
 
     *These configs are defined by the **end user**.*
 
-1. **Plugin Definition**: A structured YAML definition that describes a plugin's features, method of launch, and other configurations.
+1. **Plugin Definition**: A templatized YAML definition that describes a plugin's features, method of launch, and other configurations.
 
     *This YAML is defined by the plugin **developer**, and can be taken as a given by the end user.*
 
 The remainder of this document focuses on **Plugin Definition**.
 
 ## Plugin Definition
 
-The *plugin definition* is a YAML file (raw or wrapped in a ConfigMap) that describes the core parts of your custom plugin. It should contain the following fields:
+The *plugin definition* is a YAML file (raw or wrapped in a ConfigMap) that describes the core parts of your custom plugin. Some of the values in this YAML file are template variables that will be filled in by sonobuoy. The template will have access to several variables:
 
-| Field | Description | Example Values |
-| --- | --- | --- |
-| `name` | A name that is used to identify the plugin (e.g. in the Plugin Selection described above). | "e2e" |
-| `driver` | Sonobuoy implements *plugin drivers* that define different modes of operation.<br><br>(1) **"Job" driver**: The plugin will run on a single node (e.g. master).<br>(2) **"DaemonSet" driver**: The plugin runs on each cluster node.<br><br>You can find the implementations [here][7]. | "Job&#124;DaemonSet" |
-| `resultType` | The name of the subdirectory that this plugin's results are saved in. With a `resultType` of "e2e", results are written into `plugins/e2e/...` (within the tarball output).<br><br>This value is typically the same as the plugin `name`. | "e2e" |
-| `spec` | The Pod specification (e.g. network settings, container settings, volume definitions, etc.) | See [the parameter spec][4] below for reference. |
+| Variable | Description |
```

Comment:
To make it clearer which template variables are autofilled, could you add an additional column ("Source")?

And have the values be ("Autogenerated by Sonobuoy", "Populated from the `PluginNamespace` value in `config.json`", "Autogenerated by Sonobuoy")